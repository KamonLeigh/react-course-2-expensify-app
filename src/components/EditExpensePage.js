import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';


export class EditExpensePage extends React.Component{
    onSubmit = (expense) => {
        //this.props.dispatch(editExpense(this.props.expense.id, expense));
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/');
    }

    onClick = () => {
        this.props.startRemoveExpense({id:this.props.expense.id})
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
            <div className="page-header">
                <div className="content-container">
                    <h1>Edit Expense</h1>
                </div>
            </div>
                <div className="content-container">
                    <ExpenseForm
                    expense={this.props.expense} 
                    onSubmit={this.onSubmit}
                    />

                    <div className="button--group">
                        <button className="button--remove" onClick={this.onClick}>
                            Remove Expense
                        </button>
                        <button className="button--back">
                            <Link to="/dashboard"> Back </Link>
                        </button>
                    </div>
                </div>
        </div>
        )
    }
}


const mapDispatchToProps = (dispatch, props) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
