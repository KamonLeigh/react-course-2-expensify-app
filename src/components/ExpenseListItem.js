// Export a statrless functional component
// description, amount, creadtAt


import React from 'react';
import {Link, NavLink} from 'react-router-dom';


export  const ExpenseListItem = ({id,description, note, amount, createdAt}) => {
   return  (<div>
                <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
                <p>{amount} - {createdAt}</p>
                
            </div>)

}


export default ExpenseListItem;