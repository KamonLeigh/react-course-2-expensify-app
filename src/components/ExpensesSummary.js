import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import expenseTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expense";


 export const ExpensesSummary = ({ expensesCount, expenseTotal}) => {
     const expenseNum =  expensesCount === 1 ? "expense" : "expenses";
     console.log(expensesCount);
     const total = numeral(expenseTotal / 100).format('$0,0.00')
        return (
            <div>
                <h1>Viewing {expensesCount} {expenseNum} totalling {total}</h1>
            </div>
        )

}


const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length,
        expenseTotal: expenseTotal(visibleExpenses)
    }
}


export default connect(mapStateToProps)(ExpensesSummary)