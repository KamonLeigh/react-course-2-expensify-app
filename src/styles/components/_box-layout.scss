.box-layout {
    align-items: center;
    background: url('/img/bg.jpg');
    background-image:cover;
    display:flex;
    justify-content: center;
    height:100vh;
    width:100vw;

    &__box {
        background: fade-out(#fff, .15);
        border-radius: 3px;
        padding:$l-size $m-size;
        text-align: center;
        width: 25rem;

    }

    &__title {
        margin: 0 0 $m-size 0;
        line-height: 1;
    }
}