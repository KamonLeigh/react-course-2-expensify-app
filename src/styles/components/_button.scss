.button {
    border-radius: 5px;
    background-color: $medium-blue;
    color:#fff;
    padding:$xs-size $s-size;
    border:none;
    font-size:$font-large;
    text-transform: capitalize;
    letter-spacing: 0.1rem;
    text-decoration: none;
    display: inline-block;
    line-height:1;

}

.button--link {
    color: #fff;
    font-weight: bold;
    background-color: $dark-blue;
}

.button--remove {
    @extend .button;
    background-color: $grey;
}