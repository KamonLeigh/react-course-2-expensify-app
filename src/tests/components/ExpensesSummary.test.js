import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary }from "../../components/ExpensesSummary";
import expense from "../fixtures/expenses";


test('should handle single visible item', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expenseTotal={1000} />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle multiple visible items', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expenseTotal={100000}/>);
    expect(wrapper).toMatchSnapshot();
})