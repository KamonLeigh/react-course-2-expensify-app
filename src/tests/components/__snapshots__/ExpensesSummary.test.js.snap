// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle multiple visible items 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $1,000.00
  </h1>
</div>
`;

exports[`should handle single visible item 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $10.00
  </h1>
</div>
`;
